moveit_cartesian_plan_plugin/src/generate_cartesian_path.cpp

```diff
@@ -40,13 +41,68 @@ void GenerateCartesianPath::init()
         - Joint Model group which are necessary for checking if Way-Point is outside the IK Solution
         .
   */
-  moveit_group_ = MoveGroupPtr(new move_group_interface::MoveGroup("manipulator"));
-  kinematic_state = moveit::core::RobotStatePtr(moveit_group_->getCurrentState());
-  kinematic_state->setToDefaultValues(); 
+  selected_plan_group = 0;
+  robot_model_loader = RobotModelLoaderPtr(new robot_model_loader::RobotModelLoader("robot_description"));
+  kmodel = robot_model_loader->getModel();
+  //kmodel = robot_model::RobotModelConstPtr(new moveit::core::RobotModel(robot_model_loader->getModel()));
+
+  //group_names = kmodel->getJointModelGroupNames();
+
+  end_eff_joint_groups = kmodel->getEndEffectors();
   
 
-  const robot_model::RobotModelConstPtr &kmodel = kinematic_state->getRobotModel();
-  joint_model_group = kmodel->getJointModelGroup("manipulator");
+  ROS_INFO_STREAM("size of the end effectors is: "<<end_eff_joint_groups.size());
+
+  if (end_eff_joint_groups.empty())
+  {
+    std::vector< std::string > group_names_tmp_;
+    const moveit::core::JointModelGroup *  end_eff_joint_groups_tmp_;
+    group_names_tmp_ = kmodel->getJointModelGroupNames();
+
+    for(int i=0;i<group_names_tmp_.size();i++)
```

Comment:
This comparison, `i<group_names_tmp_.size()`, probably throws a compiler warning about comparing signed/unsigned values.  Make i of type `size_t` to avoid this.